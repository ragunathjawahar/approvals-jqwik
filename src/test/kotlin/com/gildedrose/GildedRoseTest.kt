package com.gildedrose

import net.jqwik.api.Arbitraries
import net.jqwik.api.Arbitrary
import net.jqwik.api.ForAll
import net.jqwik.api.From
import net.jqwik.api.Property
import net.jqwik.api.Provide
import org.approvaltests.Approvals

class GildedRoseTest {
  companion object {
    const val DATA_SET_SIZE = 50
  }

  @Property(
    tries = 1,
    seed = "-8556729318537930054"
  )
  fun `Aged Brie`(
    @ForAll @From("sellIns") sellIns: List<Int>,
    @ForAll @From("qualities") qualities: List<Int>,
  ) {
    // given
    val name = "Aged Brie"
    val items = sellIns.zip(qualities)
      .map { (sellIn, quality) -> Item(name, sellIn, quality) }
    val gildedRose = GildedRose(items.toTypedArray())

    // when
    gildedRose.updateQuality()

    // then
    val gildedRoseSnapshot = gildedRose
      .items
      .joinToString("\n") { it.toString() }
    Approvals.verify(gildedRoseSnapshot)
  }

  @Property(
    tries = 1,
    seed = "1002985723690070170"
  )
  fun `Backstage passes to a TAFKAL80ETC concert`(
    @ForAll @From("sellIns") sellIns: List<Int>,
    @ForAll @From("qualities") qualities: List<Int>,
  ) {
    // given
    val name = "Backstage passes to a TAFKAL80ETC concert"
    val items = sellIns.zip(qualities)
      .map { (sellIn, quality) -> Item(name, sellIn, quality) }
    val gildedRose = GildedRose(items.toTypedArray())

    // when
    gildedRose.updateQuality()

    // then
    val gildedRoseSnapshot = gildedRose
      .items
      .joinToString("\n") { it.toString() }
    Approvals.verify(gildedRoseSnapshot)
  }

  @Property(
    tries = 1,
    seed = "3398327011427988732"
  )
  fun `Sulfuras, Hand of Ragnaros`(
    @ForAll @From("sellIns") sellIns: List<Int>,
    @ForAll @From("qualities") qualities: List<Int>,
  ) {
    // given
    val name = "Sulfuras, Hand of Ragnaros"
    val items = sellIns.zip(qualities)
      .map { (sellIn, quality) -> Item(name, sellIn, quality) }
    val gildedRose = GildedRose(items.toTypedArray())

    // when
    gildedRose.updateQuality()

    // then
    val gildedRoseSnapshot = gildedRose
      .items
      .joinToString("\n") { it.toString() }
    Approvals.verify(gildedRoseSnapshot)
  }

  @Property(
    tries = 1,
    seed = "6350692656906137395"
  )
  fun `all other items`(
    @ForAll @From("sellIns") sellIns: List<Int>,
    @ForAll @From("qualities") qualities: List<Int>,
  ) {
    // given
    val name = "all other items"
    val items = sellIns.zip(qualities)
      .map { (sellIn, quality) -> Item(name, sellIn, quality) }
    val gildedRose = GildedRose(items.toTypedArray())

    // when
    gildedRose.updateQuality()

    // then
    val gildedRoseSnapshot = gildedRose
      .items
      .joinToString("\n") { it.toString() }
    Approvals.verify(gildedRoseSnapshot)
  }

  @Provide
  fun sellIns(): Arbitrary<List<Int>> =
    Arbitraries
      .integers()
      .between(0, 50)
      .list()
      .uniqueElements()
      .ofSize(DATA_SET_SIZE)

  @Provide
  fun qualities(): Arbitrary<List<Int>> =
    Arbitraries
      .integers()
      .between(-10, 60)
      .list()
      .uniqueElements()
      .ofSize(DATA_SET_SIZE)
}
