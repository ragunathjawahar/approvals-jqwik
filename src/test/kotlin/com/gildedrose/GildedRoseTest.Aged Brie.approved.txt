Aged Brie, -1, 21
Aged Brie, 5, 12
Aged Brie, 15, -1
Aged Brie, 9, -7
Aged Brie, 30, 11
Aged Brie, 24, -8
Aged Brie, 43, -4
Aged Brie, 12, 59
Aged Brie, 8, 45
Aged Brie, 3, 18
Aged Brie, 48, -9
Aged Brie, 1, -5
Aged Brie, 22, 37
Aged Brie, 17, -2
Aged Brie, 10, 32
Aged Brie, 42, 54
Aged Brie, 41, 53
Aged Brie, 0, 4
Aged Brie, 28, 0
Aged Brie, 49, 6
Aged Brie, 40, 21
Aged Brie, 36, 2
Aged Brie, 16, -6
Aged Brie, 14, 5
Aged Brie, 32, 46
Aged Brie, 2, 13
Aged Brie, 45, -3
Aged Brie, 6, 40
Aged Brie, 19, 33
Aged Brie, 38, 27
Aged Brie, 23, 50
Aged Brie, 13, 39
Aged Brie, 37, 19
Aged Brie, 29, 38
Aged Brie, 31, 1
Aged Brie, 18, 17
Aged Brie, 39, 48
Aged Brie, 47, 3
Aged Brie, 7, 26
Aged Brie, 44, 60
Aged Brie, 20, 14
Aged Brie, 33, 8
Aged Brie, 27, 28
Aged Brie, 4, 30
Aged Brie, 21, 10
Aged Brie, 26, 58
Aged Brie, 46, 41
Aged Brie, 34, 7
Aged Brie, 25, 44
Aged Brie, 11, 43
